.github/scripts/transform_dbt_test_results.py

```diff
@@ -0,0 +1,1082 @@
+                for col_group_indexes, col_metadata in column_groups.items():
+                    if column_letter in col_group_indexes:
+                        cell_style = col_metadata["style"]
+                if cell_style:
+                    cell.style = cell_style
+                data_row.append(cell)
+
+                # Check if this cell is longer than the longest cell we've seen
+                # so far, and adjust the column dimensions accordingly
+                column_letter = openpyxl.utils.get_column_letter(col_idx)
+                column_widths[column_letter] = max(
+                    column_widths.get(column_letter, 0), len(cell_str)
+                )
+
+            data_rows.append(data_row)
+
+        # Update column widths so that they fit the longest column
+        for (
+            column_letter,
+            column_width,
+        ) in column_widths.items():
+            # Pad with an extra two characters to account for the fact that
+            # non-monospace fonts do not have consistent character widths,
+            # and set a hard limit of 75 characters so no one field takes over
+            # the viewport of the spreadsheet
+            width = min(column_width + 2, 75)
+            sheet.column_dimensions[column_letter].width = width
+
+        # Add filters to fixed columns (i.e. columns that appear in every sheet
+        # in the same position)
+        fixed_field_indexes = self.fixed_field_indexes
+        sheet_max_row_idx = data_header_idx + len(data_rows)
+        min_fixed_idx = f"{fixed_field_indexes[0]}{data_header_idx}"
+        max_fixed_idx = f"{fixed_field_indexes[-1]}{sheet_max_row_idx}"
+        fixed_field_range = f"{min_fixed_idx}:{max_fixed_idx}"
+        sheet.auto_filter.ref = fixed_field_range
+
+        # Add the data to the sheet. This should be one of the last steps in
+        # this function, since write-only sheets require all formatting to be
+        # set before data is added
+        sheet.append(title_row)
+        sheet.append(subtitle_row)
+        sheet.append(header_row)
+        for data_row in data_rows:
+            sheet.append(data_row)
+
+        # Merge cells in the grouping headers. This approach is a bit of a hack
+        # since merged cells are not fully supported in write-only workbooks,
+        # hence why it takes place _after_ rows have been added to the sheet
+        # whereas most formatting options for write-only workbooks need to
+        # happen _before_ data is added. See here for details:
+        # https://stackoverflow.com/a/66159254
+        for cell_range in merged_cell_range:
+            sheet.merged_cells.ranges.add(cell_range)
+
+    @property
+    def debugging_fieldnames(self) -> typing.List[str]:
+        """Get a list of fieldnames (e.g. ["foo", "bar"]) for fields that
+        are used for debugging."""
+        return self._filter_for_existing_fieldnames(self._debugging_fieldnames)
```

Comment:
This property function and all of the ones that follow were copied over as-is from the old `FailedTestGroup` object.